core/src/main/scala/kafka/server/KafkaServer.scala

```diff
@@ -113,10 +113,12 @@ class KafkaServer(val config: KafkaConfig, time: Time = Time.SYSTEM, threadNameP
 
   val brokerState: BrokerState = new BrokerState
 
-  var apis: KafkaApis = null
+  var dataPlaneApis: KafkaApis = null
+  var controlPlaneApis: KafkaApis = null
```

Comment:
KafkaApis contains the logic to handle request and itself should contain no state (other than those state in its constructor parameters). So it seems weird to have two KafkaApis. Currently KafkaApis only uses `requestChannel` with together with `RequestChannel.Request`. One reasonable approach is to put the `requestChannel` in `RequestChannel.Request` similar to the existing `RequestChannel.Request.processor`. So that we no longer need two KafkaApis instance. I am not sure this is the best alternative approach though.